### PR TITLE
Issue108/gpl 815 migration off travis

### LINKED ISSUE #108: GPL-815 Migration off travis
This represents the sm_workflow_lims work required for https://github.com/sanger/General-Backlog-Items/issues/109

Original issue:
# Background

Travis are have changed their open-source free tier, which places heavy restrictions on our needs. We are currently on a paid plan, but are looking to migrate off.

I have made notes about the changes made to migrate Sequencescape here:
https://github.com/JamesGlover/travis_to_gh_actions

# Migrated
- [x] Sequencescape
- [x] Warren

# Still on travis, but some actions
These may already have all their actions migrated, although I think its mostly docker build files.
They should have all actions migrated, and be removed from travis
- [x] Baracoda
- [ ]  lighthouse-ui
- [ ] samples_extraction
- [ ] print_my_barcode

# Repo to migrate
- [ ] Traction
- [x] Asset Audits
- [ ] Traction-ui
- [ ] Quanthub
- [x] Sm_workflow_lims
- [ ] sprint
- [ ] eln_pmb_bridge [Can't see on travis.com]
- [ ] mixtio
- [x] gatekeeper
- [x] baracoda
- [ ] traction-service-automated-releases ? (What is this?)
- [x] traction-service
- [ ] event_warehouse_ukbb
- [x] sanger_barcode_format
- [x] event_warehouse
- [x] unified_warehouse
- [x] limber
- [x] pmb-client
- [x] labwhere

# Probably don't migrate
- [ ] json_api_client 
This is a fork of another gem, which brings in some fixes. We should focus of migrating to the latest version, rather than maintaining a fork.

# Repos without CI
These repos currently have no CI solution. In some cases its because one probably isn't warranted.
## CI needed?
- [ ] sequencescape-client-api
- [ ] sprint_client
- [ ] hca-print-tool
- [ ] ftpcmd
- [ ] warehouse_two

## CI probably not needed
- [ ] storelight-sql
- [ ] config
- [ ] .github
- [ ] beckman-robot-scripts
- [ ] deployment
- [ ] provision
- [ ] usg_scripts
- [ ] ss_cherrypicking (Empty repo)
- [ ] plate_barcode [ Migrate to baracoda instead ]
- [ ] General backlog items
- [ ] axlsx [Migrate to community fork instead of maintaining our own]
- [ ] label_printer [Yana's tool, mostly replaced by steve's recent stuff, not sure how you'd ci the tests]
- [ ] sequencescape_search [I don't think we use this?]
- [ ] sanger_barcode [Legacy barcode printing]
- [ ] irods_reader [Not using functionality]
- [ ] psd_logger [No longer used?]

# Repos already on GH actions
These repos either never used travis, or were migrated a while ago. I'm judging this by the presence of a .github/workflows folder
- [x] Record Loader(Never used travis)
- [x] Crawler
- [x] storelight
- [x] lighthouse
- [x] wrangler
- [x] stan-client
- [x] stan-core
- [x] automanual

### PR DESCRIPTION
**Pull request by JamesGlover**
Closes #108

Changes proposed in this pull request:

* Migrates from travis to GH actions
* Adds rubocop (Probably want to ignore the commits that begin Rubocop, with perhaps exception of Line length)

### CODE REVIEW

File: .github/workflows/ruby_ci.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+
```
Comment (pjvv) [inline .github/workflows/ruby_ci.yml:1]:
```suggestion
```

File: .github/workflows/ruby_build.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+name: Ruby Build
+
+on:
+  release:
+    types: published
```
Comment (pjvv) [inline .github/workflows/ruby_build.yml:5]: Does this also happen for drafts?

File: .github/workflows/ruby_ci.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+name: Ruby Lint and Test
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  test:
+
```
Comment (pjvv) [inline .github/workflows/ruby_ci.yml:10]:
```suggestion
```

File: .github/workflows/ruby_ci.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                            # example if you need more control over bundler.
+    - name: Setup database
+      run: bundle exec rake db:setup
+    - name: Test & publish code coverage
+      uses: paambaati/codeclimate-action@v2.7.5
+      env:
+        CC_TEST_REPORTER_ID:  ${{ secrets.CC_TEST_REPORTER_ID || '5e399530a457db7a41cd5785ce0536b79e9022b3c2d4382f101310b3b166eb38' }}
+      with:
+        coverageCommand: bundle exec rspec
+  lint:
```
Comment (pjvv) [inline .github/workflows/ruby_ci.yml:44]: Does it matter to test before linting?

File: config/data/workflows.rb [hunk 5]
```diff
@@ -1,35 +1,56 @@
...
-      {name: 'Fluidigm CGP 96:96',               has_comment: false, reportable: false, initial_state_name: 'in_progress' },
-      {name: 'Storage',                          has_comment: false, reportable: false, initial_state_name: 'in_progress' },
-      {name: 'Sequenom prep (Independent)',      has_comment: false, reportable: false, initial_state_name: 'in_progress' },
-      {name: 'Sequenom prep (QC)',               has_comment: false, reportable: false, initial_state_name: 'in_progress' },
-      {name: 'Other',                            has_comment: true,  reportable: false, initial_state_name: 'in_progress' },
-      {name: 'Volume Check',                     has_comment: false, reportable: false, initial_state_name: 'in_progress' },
-      {name: 'Fluidigm DDD 96:96',               has_comment: false, reportable: false, initial_state_name: 'in_progress' },
-      {name: 'DNA extraction QS',                has_comment: false, reportable: false, initial_state_name: 'in_progress', active: false }
+      { name: 'Arrival and issue',                has_comment: false, reportable: false,
```
Comment (pjvv) [inline config/data/workflows.rb:4]: Weird formatting, I would expect one below the other:
```
{ 
  name: 'DNA and RNA Extraction Q3',        
  has_comment: false, reportable: false,
  initial_state_name: 'in_progress' 
},
```

**Review by pjvv — APPROVED** [review]
Looks like the code got a bath! Nice work.

File: .github/workflows/ruby_build.yml (hunk 2 above)
Comment (JamesGlover) [reply .github/workflows/ruby_build.yml:5]: Not drafts, but does work for pre-releases. I kind of assumed the whole point of a draft release was that it wasn't ready yet and thus you wouldn't want to build it. I don't tend to use them myself though, so maybe I'm missing a use case.

File: .github/workflows/ruby_ci.yml (hunk 4 above)
Comment (JamesGlover) [reply .github/workflows/ruby_ci.yml:42]: They are separate jobs, so happen in parallel.

**Comment by JamesGlover** [comment]
> Looks like the code got a bath! Nice work.

I'd like to claim responsibility, but it was mostly rubocop.

File: .github/workflows/ruby_build.yml (hunk 2 above)
Comment (pjvv) [reply .github/workflows/ruby_build.yml:5]: No sorry I meant pre-release!

File: config/data/workflows.rb (hunk 5 above)
Comment (JamesGlover) [reply config/data/workflows.rb:4]: Rubocop's auto-formatting isn't quite as opinionated as black's, I think here it was trying to maintain the original
indenting from when it was all on a single line... but I'm actually not sure why. I'll clean that up manually.

**Review by pjvv — APPROVED** [review]

File: config/data/workflows.rb (hunk 5 above)
Comment (JamesGlover) [reply config/data/workflows.rb]: Fixed it up now. It did push up the module/method length, but in that case I was happy to add a cop disable, as its essentially config. (Could load it from a yaml, but I can only chase down so many rabbit holes)

Comment (pjvv) [reply config/data/workflows.rb]: No probs.

**Review by pjvv — APPROVED** [review]

**Pull request merged**